app/callbacks/views/sns.py

```diff
@@ -0,0 +1,417 @@
+import datetime
+from functools import lru_cache
+from itertools import chain
+import json
+import logging
+import re
+import sys
+
+import boto3
+from flask import abort, jsonify, current_app, request
+from lxml.etree import fromstring as etree_fromstring, ParseError
+import requests
+
+import validatesns
+
+from dmutils.email.dm_notify import DMNotifyClient
+from dmutils.email.exceptions import EmailError
+from dmutils.timing import logged_duration_for_external_request as log_external_request, logged_duration
+
+from app.callbacks import callbacks
+from app.clam import get_clamd_socket
+
+
+@callbacks.route('/')
+@callbacks.route('')
+def callbacks_root():
+    return jsonify(status='ok'), 200
+
+
+def _get_request_body_json():
+    if request.content_type.split(";")[0] != "application/json":  # allowing for an encoding following the semicolon
+        abort(400, "Unexpected Content-Type, expecting 'application/json'")
+
+    data = request.get_json()
+
+    if data is None:
+        abort(400, "Invalid JSON; must be a valid JSON object")
+
+    return data
+
+
+def _handle_subscription_confirmation(body_json, supported_topic_name):
+    topic_arn = body_json["TopicArn"]
+    topic_name = topic_arn.split(":")[-1]
+    # check it's for the right topic
+    if topic_name != supported_topic_name:
+        current_app.logger.warning(
+            "Received SubscriptionConfirmation request for unrecognized topic name {topic_name} in {topic_arn}",
+            extra={
+                "topic_name": topic_name,
+                "topic_arn": topic_arn,
+            },
+        )
+        abort(400, f"Unrecognized topic name {topic_name!r}")
```

Comment:
Is the` __repr__` formatting just to put quotes around it? `topic_name` will only ever be a string, right?